autogpt_platform/backend/backend/blocks/github/triggers.py

```diff
@@ -0,0 +1,132 @@
+import logging
+
+from pydantic import BaseModel
+
+from backend.data.block import (
+    Block,
+    BlockCategory,
+    BlockOutput,
+    BlockSchema,
+    BlockWebhookConfig,
+)
+from backend.data.model import SchemaField
+
+from ._auth import (
+    TEST_CREDENTIALS,
+    TEST_CREDENTIALS_INPUT,
+    GithubCredentialsField,
+    GithubCredentialsInput,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class GitHubTriggerBase:
+    class Input(BlockSchema):
+        credentials: GithubCredentialsInput = GithubCredentialsField("repo")
+        repo: str = SchemaField(
+            description="Repository to subscribe to",
+            placeholder="{owner}/{repo}",
+        )
+        payload: dict = SchemaField(hidden=True)
+
+    class Output(BlockSchema):
+        event: str = SchemaField(description="The event that triggered the webhook")
+        payload: dict = SchemaField(description="Full payload of the event")
+        sender: dict = SchemaField(
+            description="Object representing the user who triggered the event"
+        )
+        error: str = SchemaField(
+            description="Error message if the payload could not be processed"
+        )
+
+    def run(self, input_data: Input, **kwargs) -> BlockOutput:
+        yield "event", input_data.payload["action"]
+        yield "payload", input_data.payload
+        yield "sender", input_data.payload["sender"]
+
+
+class GithubPullRequestTriggerBlock(GitHubTriggerBase, Block):
+    class Input(GitHubTriggerBase.Input):
+        class EventsFilter(BaseModel):
+            """
+            https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request
+            """
+
+            opened: bool = False
+            edited: bool = False
+            closed: bool = False
+            reopened: bool = False
+            synchronize: bool = False
+            assigned: bool = False
+            unassigned: bool = False
+            labeled: bool = False
+            unlabeled: bool = False
+            converted_to_draft: bool = False
+            locked: bool = False
+            unlocked: bool = False
+            enqueued: bool = False
+            dequeued: bool = False
+            milestoned: bool = False
+            demilestoned: bool = False
+            ready_for_review: bool = False
+            review_requested: bool = False
+            review_request_removed: bool = False
+            auto_merge_enabled: bool = False
+            auto_merge_disabled: bool = False
```

Comment:
lol what happens if all of them are false

Reply:
It doesn't for check for that (yet), probably should though:
https://github.com/Significant-Gravitas/AutoGPT/blob/aa12a5a6f3ec02cc323dbd53dbb77b1c15966a40/autogpt_platform/backend/backend/integrations/webhooks/graph_lifecycle_hooks.py#L129-L140

Reply:
Added